orchestra/todos/auth.py

```diff
@@ -2,17 +2,25 @@
 
 from orchestra.models import Worker
 from orchestra.models import Task
+from orchestra.models import Todo
+from orchestra.models import TodoQA
 
 
 class IsAssociatedWithTodosProject(permissions.BasePermission):
     """
     Ensures that a user's worker is accoiated with the todo's project.
     """
 
-    def has_object_permission(self, request, view, todo):
+    def has_object_permission(self, request, view, obj):
         worker = Worker.objects.get(user=request.user)
-        project = todo.task.project
+        if isinstance(obj, Todo):
+            project = obj.task.project
+        elif isinstance(obj, TodoQA):
+            project = obj.todo.task.project
+        else:
+            project = None
         return (
+            project and
```

Comment:
do you need `project and`, and if you do, should there be parenthesis around the next two boolean checks?